hazelcast/src/main/java/com/hazelcast/map/impl/SizeEstimatorFactory.java

```diff
@@ -16,45 +16,44 @@
 
 package com.hazelcast.map.impl;
 
-
 import com.hazelcast.config.InMemoryFormat;
 import com.hazelcast.config.NearCacheConfig;
+import com.hazelcast.map.impl.nearcache.NearCacheRecord;
 import com.hazelcast.map.impl.nearcache.NearCacheSizeEstimator;
 
 import static com.hazelcast.config.InMemoryFormat.BINARY;
 
 /**
  * Static factory methods for various size estimators.
  */
-public final class SizeEstimators {
+public final class SizeEstimatorFactory {
 
     /**
      * Returns zero for all estimations.
      */
     private static final SizeEstimator ZERO_SIZE_ESTIMATOR = new ZeroSizeEstimator();
 
-    private SizeEstimators() {
+    private SizeEstimatorFactory() {
     }
 
     public static SizeEstimator createMapSizeEstimator(InMemoryFormat inMemoryFormat) {
         if (BINARY.equals(inMemoryFormat)) {
             return new BinaryMapSizeEstimator();
-        } else {
-            return ZERO_SIZE_ESTIMATOR;
         }
+        return ZERO_SIZE_ESTIMATOR;
     }
 
-    public static SizeEstimator createNearCacheSizeEstimator(NearCacheConfig nearCacheConfig) {
+    @SuppressWarnings("unchecked")
```

Comment:
The generics are now an implementation details of the factory, so I'm happy with the suppression here. We could of course solve this down to the bottom, but I think it's good enough this way.